### PR TITLE
Optimized image scale down performance

### PR DESCRIPTION
**Pull request by omichel**
Avoiding the use of float and conversion from float to int allow us to speed up the image scale down algorithm. Before this PR, it took 127 ms on my PC, with this PR, it takes 69 ms.

### CODE REVIEW

**Comment by omichel** [comment]
test me

**Comment by fabienrohrer** [comment]
Just to inform you also that there is a huge difference if Webots is built in debug or in release mode. The compiler optimizer do a very good job on this loop.

File: src/webots/maths/WbImage.cpp [hunk 1]
```diff
@@ -19,36 +19,33 @@
...
+        unsigned char acc = 0;
         for (int u = -1; u < 2; ++u) {
-          const float x = CLAMP(widthRatio * i + u * xBlurRadius, 0, mWidth - 1);
+          const int x = CLAMP(mWidth * i / width + u * xBlurRadius, 0, mWidth - 1);
           for (int v = -1; v < 2; ++v) {
-            const float y = CLAMP(heightRatio * j + v * yBlurRadius, 0, mHeight - 1);
+            const int y = CLAMP(mHeight * j / height + v * yBlurRadius, 0, mHeight - 1);
             const int srcIndex = CLAMP((int)(x + y * mWidth) * channels + c, 0, srcSize);
-            pixels[destIndex] += (unsigned char)(gaussBlur[u + 1][v + 1] * mData[srcIndex]);
+            acc += gaussBlur[u + 1][v + 1] * mData[srcIndex] / 16;
```
Comment (fabienrohrer) [inline src/webots/maths/WbImage.cpp]:
```suggestion
            acc += gaussBlur[u + 1][v + 1] * mData[srcIndex] >> 4;
```

Comment (omichel) [reply src/webots/maths/WbImage.cpp]: It makes no difference as the compiler automatically translates `/ 16` into `>> 4` for integers. I measured the performance of both cases, and cannot observe any difference.

Comment (omichel) [reply src/webots/maths/WbImage.cpp]: And I believe `/ 16` is more readable.

Comment (fabienrohrer) [reply src/webots/maths/WbImage.cpp]: Thank you for the test. It's indeed quite obvious that the compiler can deal with this.

**Comment by omichel** [comment]
My performance tests were measured in release mode.

**Comment by omichel** [comment]
👍

**Comment by omichel** [comment]
Thank you for the tests.

**Pull request merged**